rmw_cyclonedds_cpp/src/serdata.cpp

```diff
@@ -244,10 +251,10 @@ static struct ddsi_serdata * serdata_rmw_from_iox(
   const struct ddsi_sertype * typecmn,
   enum  ddsi_serdata_kind kind, void * sub, void * iox_buffer)
 {
-  static_cast<void>(sub);  // unused
   const struct sertype_rmw * type = static_cast<const struct sertype_rmw *>(typecmn);
   auto d = std::make_unique<serdata_rmw>(type, kind);
   d->iox_chunk = iox_buffer;
+  d->iox_subscriber = sub;
   return d.release();
```

Comment:
Why create a unique pointer if we release it to return a raw pointer after two instructions that cannot throw (i.e. there is no benefit from RAII here)? We could create a raw pointer instead which would be slightly more efficient.

Reply:
This pattern is used in multiple places in the code base, I don't want to change this in the scope of this MR, but I have wrapped the creation of new serdata in a try-catch block to catch `std::bad_alloc`